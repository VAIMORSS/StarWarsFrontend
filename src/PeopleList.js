export const PeopleList =[
    { id:1, name:'Luke Skywalker'},
    { id:2, name:'C-3PO'},
    { id:3, name:'R2-D2'},
    { id:4, name:'Darth Vader'},
    { id:5, name:'Leia Organa'},
      
]