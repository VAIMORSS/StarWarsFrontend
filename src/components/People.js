import React, { Component } from 'react'
import PropTypes from "prop-types";
import { connect } from "react-redux";
import SinglePerson from './singlePerson';
import { getPeopleList } from '../actions/peopleActions';
import "./../App.css";
import logo from './../img/logo.png'


class People extends Component {


    constructor(props) {
        super(props);
        this.state={
            search:''
        }
    }

    componentDidMount(){
        this.props.getPeopleList();
    }

    handleChange(event){
        this.setState({
            search:event.target.value
        })
    }

    render() {
        const {people} = this.props.people; 
        
        let filteredPeople = people.filter(
            (person)=>{
                return person.name.toLowerCase().indexOf(this.state.search.toLowerCase()) !== -1; 
            }
        );


        return (
            <div className="people container">
                <img className="logo" src={logo} alt="logo" />
                <input
                    type="text"
                    className="input-field col s12 searchBar"
                    value={this.state.searchString}
                    ref="search"
                    onChange={this.handleChange.bind(this)}
                    placeholder="Search Here"
                />
                <div className="card-deck">
                    {filteredPeople.map(person=>
                            <div key={person.id}>
                                <SinglePerson key={person.id} Person={person} />
                            </div>
                    )}
                </div>
                
            </div>
        )
    }
}

People.propTypes={
    people:PropTypes.object.isRequired,
    getPeopleList:PropTypes.func.isRequired
}

const mapStateToProps = state =>({
    people:state.people
})

export default connect(
    mapStateToProps,
    {getPeopleList}
)(People);

