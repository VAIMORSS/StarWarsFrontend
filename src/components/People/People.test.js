import React from 'react'
import { mount } from 'enzyme';
import People from '.';
import configureStore  from 'redux-mock-store';
import { PeopleList } from '../../assets/PeopleList';
import { Provider } from 'react-redux';

describe('People Test', () => {
     const test=()=>(dispatch)=>dispatch({
         type:'lala',
         value:{}
     })
    const initialStore = {
        people:{people:PeopleList},
        getPeopleList:test
    };
    const mockStore = configureStore()
    let store,container;

    beforeEach(()=>{
        store = mockStore(initialStore);
        container = mount(
        <Provider store={store}> <People  /></Provider>
       )  
    })

    it('should render correctly in "debug" mode', () => {
        expect(container).toMatchSnapshot();
    })

    it('on text in the search bar changed the search string state should change', () => {
        
        const input = container.find('input#searchBar');
        input.simulate('focus');
        input.simulate('change', { target: { value: 'Changed' } });
        input.simulate('keyDown', {
            which: 27,
            target: {
                blur() {
                    // Needed since <EditableText /> calls target.blur()
                    input.simulate('blur');
                },
            },
        });
        
        expect(container.find('People').instance().state.search).toEqual('Changed');
    })
});