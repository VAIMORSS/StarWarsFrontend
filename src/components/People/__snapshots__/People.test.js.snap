// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People Test should render correctly in "debug" mode 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
   
  <Connect(People)>
    <People
      getPeopleList={[Function]}
      people={
        Object {
          "people": Array [
            Object {
              "id": 1,
              "name": "Luke Skywalker",
            },
            Object {
              "id": 2,
              "name": "C-3PO",
            },
            Object {
              "id": 3,
              "name": "R2-D2",
            },
            Object {
              "id": 4,
              "name": "Darth Vader",
            },
            Object {
              "id": 5,
              "name": "Leia Organa",
            },
            Object {
              "id": 6,
              "name": "Owen Lars",
            },
            Object {
              "id": 7,
              "name": "Beru Whitesun lars",
            },
            Object {
              "id": 8,
              "name": "R5-D4",
            },
            Object {
              "id": 9,
              "name": "Biggs Darklighter",
            },
            Object {
              "id": 10,
              "name": "Leia Organa",
            },
          ],
        }
      }
    >
      <div
        className="people container"
      >
        <img
          alt="logo"
          className="logo"
          src="logo.png"
        />
        <input
          className="input-field col s12 searchBar"
          id="searchBar"
          onChange={[Function]}
          placeholder="Search Here"
          type="text"
        />
        <div
          className="card-deck"
        >
          <div
            key="1"
          >
            <SinglePerson
              Person={
                Object {
                  "id": 1,
                  "name": "Luke Skywalker",
                }
              }
              key="1"
            >
              <div
                className="card"
              >
                <img
                  alt="Card image cap"
                  className="card-img-top avatar"
                  src="https://starwarsblog.starwars.com/wp-content/uploads/2017/01/sw-the-last-jedi-tall-1200x630.jpg"
                />
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Luke Skywalker
                  </h5>
                </div>
                <div
                  className="card-footer"
                >
                  <button
                    className="btn btn-dark 1"
                    onClick={[Function]}
                  >
                    More Info
                  </button>
                </div>
              </div>
            </SinglePerson>
          </div>
          <div
            key="2"
          >
            <SinglePerson
              Person={
                Object {
                  "id": 2,
                  "name": "C-3PO",
                }
              }
              key="2"
            >
              <div
                className="card"
              >
                <img
                  alt="Card image cap"
                  className="card-img-top avatar"
                  src="https://starwarsblog.starwars.com/wp-content/uploads/2017/01/sw-the-last-jedi-tall-1200x630.jpg"
                />
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    C-3PO
                  </h5>
                </div>
                <div
                  className="card-footer"
                >
                  <button
                    className="btn btn-dark 2"
                    onClick={[Function]}
                  >
                    More Info
                  </button>
                </div>
              </div>
            </SinglePerson>
          </div>
          <div
            key="3"
          >
            <SinglePerson
              Person={
                Object {
                  "id": 3,
                  "name": "R2-D2",
                }
              }
              key="3"
            >
              <div
                className="card"
              >
                <img
                  alt="Card image cap"
                  className="card-img-top avatar"
                  src="https://starwarsblog.starwars.com/wp-content/uploads/2017/01/sw-the-last-jedi-tall-1200x630.jpg"
                />
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    R2-D2
                  </h5>
                </div>
                <div
                  className="card-footer"
                >
                  <button
                    className="btn btn-dark 3"
                    onClick={[Function]}
                  >
                    More Info
                  </button>
                </div>
              </div>
            </SinglePerson>
          </div>
          <div
            key="4"
          >
            <SinglePerson
              Person={
                Object {
                  "id": 4,
                  "name": "Darth Vader",
                }
              }
              key="4"
            >
              <div
                className="card"
              >
                <img
                  alt="Card image cap"
                  className="card-img-top avatar"
                  src="https://starwarsblog.starwars.com/wp-content/uploads/2017/01/sw-the-last-jedi-tall-1200x630.jpg"
                />
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Darth Vader
                  </h5>
                </div>
                <div
                  className="card-footer"
                >
                  <button
                    className="btn btn-dark 4"
                    onClick={[Function]}
                  >
                    More Info
                  </button>
                </div>
              </div>
            </SinglePerson>
          </div>
          <div
            key="5"
          >
            <SinglePerson
              Person={
                Object {
                  "id": 5,
                  "name": "Leia Organa",
                }
              }
              key="5"
            >
              <div
                className="card"
              >
                <img
                  alt="Card image cap"
                  className="card-img-top avatar"
                  src="https://starwarsblog.starwars.com/wp-content/uploads/2017/01/sw-the-last-jedi-tall-1200x630.jpg"
                />
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Leia Organa
                  </h5>
                </div>
                <div
                  className="card-footer"
                >
                  <button
                    className="btn btn-dark 5"
                    onClick={[Function]}
                  >
                    More Info
                  </button>
                </div>
              </div>
            </SinglePerson>
          </div>
          <div
            key="6"
          >
            <SinglePerson
              Person={
                Object {
                  "id": 6,
                  "name": "Owen Lars",
                }
              }
              key="6"
            >
              <div
                className="card"
              >
                <img
                  alt="Card image cap"
                  className="card-img-top avatar"
                  src="https://starwarsblog.starwars.com/wp-content/uploads/2017/01/sw-the-last-jedi-tall-1200x630.jpg"
                />
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Owen Lars
                  </h5>
                </div>
                <div
                  className="card-footer"
                >
                  <button
                    className="btn btn-dark 6"
                    onClick={[Function]}
                  >
                    More Info
                  </button>
                </div>
              </div>
            </SinglePerson>
          </div>
          <div
            key="7"
          >
            <SinglePerson
              Person={
                Object {
                  "id": 7,
                  "name": "Beru Whitesun lars",
                }
              }
              key="7"
            >
              <div
                className="card"
              >
                <img
                  alt="Card image cap"
                  className="card-img-top avatar"
                  src="https://starwarsblog.starwars.com/wp-content/uploads/2017/01/sw-the-last-jedi-tall-1200x630.jpg"
                />
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Beru Whitesun lars
                  </h5>
                </div>
                <div
                  className="card-footer"
                >
                  <button
                    className="btn btn-dark 7"
                    onClick={[Function]}
                  >
                    More Info
                  </button>
                </div>
              </div>
            </SinglePerson>
          </div>
          <div
            key="8"
          >
            <SinglePerson
              Person={
                Object {
                  "id": 8,
                  "name": "R5-D4",
                }
              }
              key="8"
            >
              <div
                className="card"
              >
                <img
                  alt="Card image cap"
                  className="card-img-top avatar"
                  src="https://starwarsblog.starwars.com/wp-content/uploads/2017/01/sw-the-last-jedi-tall-1200x630.jpg"
                />
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    R5-D4
                  </h5>
                </div>
                <div
                  className="card-footer"
                >
                  <button
                    className="btn btn-dark 8"
                    onClick={[Function]}
                  >
                    More Info
                  </button>
                </div>
              </div>
            </SinglePerson>
          </div>
          <div
            key="9"
          >
            <SinglePerson
              Person={
                Object {
                  "id": 9,
                  "name": "Biggs Darklighter",
                }
              }
              key="9"
            >
              <div
                className="card"
              >
                <img
                  alt="Card image cap"
                  className="card-img-top avatar"
                  src="https://starwarsblog.starwars.com/wp-content/uploads/2017/01/sw-the-last-jedi-tall-1200x630.jpg"
                />
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Biggs Darklighter
                  </h5>
                </div>
                <div
                  className="card-footer"
                >
                  <button
                    className="btn btn-dark 9"
                    onClick={[Function]}
                  >
                    More Info
                  </button>
                </div>
              </div>
            </SinglePerson>
          </div>
          <div
            key="10"
          >
            <SinglePerson
              Person={
                Object {
                  "id": 10,
                  "name": "Leia Organa",
                }
              }
              key="10"
            >
              <div
                className="card"
              >
                <img
                  alt="Card image cap"
                  className="card-img-top avatar"
                  src="https://starwarsblog.starwars.com/wp-content/uploads/2017/01/sw-the-last-jedi-tall-1200x630.jpg"
                />
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Leia Organa
                  </h5>
                </div>
                <div
                  className="card-footer"
                >
                  <button
                    className="btn btn-dark 10"
                    onClick={[Function]}
                  >
                    More Info
                  </button>
                </div>
              </div>
            </SinglePerson>
          </div>
        </div>
      </div>
    </People>
  </Connect(People)>
</Provider>
`;
